app/models/invite.rb

```diff
@@ -182,7 +182,9 @@ def redeem(email: nil, username: nil, name: nil, password: nil, user_custom_fiel
 
   def self.redeem_from_email(email)
     invite = Invite.find_by(email: Email.downcase(email))
-    InviteRedeemer.new(invite: invite, email: invite.email).redeem if invite
+    if invite.present? && invite.redeemable?
```

Comment:
```suggestion
    if invite.present?
```

I don't think we need this check here anymore?

Reply:
I left it here because there's no harm in checking early and aborting?

Reply:
but it means we continue to duplicate the logic of whether we can redeem an invite or not in multiple places. This condition will no longer be useful if we add on more conditions to when an invite can be redeemed in the future.

Reply:
Note that `redeemable?` is a method which is checking for many things and in future if the condition changes it will be reflected here as well.

https://github.com/discourse/discourse/blob/20462cf2b5c142ff0b5721a77c905fbcf33465e4/app/models/invite.rb#L73-L75

But if you still think it should be removed from here then I'll do that and also remove it from `redeem` method.

Reply:
Everything should be centralized in 

https://github.com/discourse/discourse/blob/c1cdb6bc51873afa1c351be6bbe878616ccd341b/app/models/invite_redeemer.rb#L84-L99

since it is the service object which we use to redeem an invite. I guess my biggest question is why do we need `Invite#reedemable?` and `InviteRedeemer#can_redeem_invite?`. Both methods seem to attempt to do the same thing and we should just centralize the logic so that further developers don't trip on a bug like this again.

Reply:
`Invite#reedemable?` is also being used to verify the invite record without trying to redeem it for example here:

https://github.com/discourse/discourse/blob/fa8cd629f1ad4c64308e3e5c1d93f3ae78b2815a/app/services/user_activator.rb#L36

`InviteRedeemer#can_redeem_invite?` is doing some things which we need to check at the time of redeeming invite and might not make sense when we just want to validate the invite.

Reply:
I guess we can always create a `InviteRedeemer.redeemable?(invite)` method but that might be out of scope for this change.